### PR TITLE
feat: typed element groups and per-element variables in generated API

### LINKED ISSUE #215: feat: extend generated API with typed elements and per-node variables
## Summary

Make the enriched domain model (from #213) and sequence flow data (from #214) visible in the generated API. The internal restructuring creates value only when it surfaces in the code that users interact with.

## Current State

The generated API has a flat `Elements` object with all flow nodes as undifferentiated string constants:

```kotlin
object Elements {
    const val START_EVENT_SUBMIT_FORM: String = "StartEvent_SubmitForm"
    const val ACTIVITY_SEND_MAIL: String = "Activity_SendMail"
    const val GATEWAY_EXCLUSIVE_1: String = "Gateway_Exclusive1"
    // all mixed together, no type grouping
}

object Variables {
    const val EMAIL: String = "email"
    const val SUBSCRIPTION_ID: String = "subscriptionId"
    // flat global list, no association to elements
}
```

### Typed element groups

```kotlin
object Elements {
    object Tasks {
        const val ACTIVITY_SEND_MAIL: String = "Activity_SendMail"
        const val ACTIVITY_CONFIRM_REGISTRATION: String = "Activity_ConfirmRegistration"
    }
    object Gateways {
        const val GATEWAY_EXCLUSIVE_1: String = "Gateway_Exclusive1"
    }
    object Events {
        const val START_EVENT_SUBMIT_FORM: String = "StartEvent_SubmitForm"
        const val END_EVENT_DONE: String = "EndEvent_Done"
    }
}
```

### Variables per element

```kotlin
object Variables {
    // Global list (backward compatible)
    const val EMAIL: String = "email"
    const val SUBSCRIPTION_ID: String = "subscriptionId"

    // Per-element grouping (new)
    object ActivitySendMail {
        const val EMAIL: String = "email"
    }
    object ActivityConfirmRegistration {
        const val SUBSCRIPTION_ID: String = "subscriptionId"
    }
}
```

### Optional: Graph utility methods

```kotlin
object ProcessFlow {
    fun successorsOf(elementId: String): List<String> = when(elementId) {
        "StartEvent_SubmitForm" -> listOf("Activity_SendMail")
        "Activity_SendMail" -> listOf("Gateway_Exclusive1")
        // ...
        else -> emptyList()
    }
}
```

## Technical Notes

- Requires #213 (FlowNodeType on FlowNodeDefinition for grouping, variables per element)
- Requires #214 (sequence flows for graph utilities)
- This is a **breaking change** to the generated API structure (Elements sub-grouping). Consider:
  - Opt-in via configuration flag (`groupElements = true`)
  - Or new major version
- The flat `Elements` object should remain available as a backward-compatible option
- Per-element variables are additive (new nested objects in Variables), not breaking

Depends on #213 and #214.

### PR DESCRIPTION
**Pull request by emaarco**
## Summary

Closes #215 (partially — ProcessFlow utilities deferred to follow-up).

- `Elements` now contains nested group sub-objects (`Tasks`, `Events`, `Gateways`, `Containers`) alongside the existing flat constants — fully backward compatible
- `Variables` now contains per-element nested objects for each flow node that carries variables (e.g. `Variables.ActivitySendMail.SUBSCRIPTION_ID`)
- Added `StringUtils.toCamelCase()` for converting flow node IDs to valid class/object names

## Test plan

- [x] `KotlinApiBuilderTest` snapshot updated and passing
- [x] `JavaApiBuilderTest` snapshot updated and passing (including hyphen-in-ID edge case)
- [x] `./gradlew :bpmn-to-code-core:test` — all 102+ tests green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: bpmn-to-code-core/src/main/kotlin/io/github/emaarco/bpmn/adapter/outbound/codegen/builder/JavaApiBuilder.kt [hunk 1]
```diff
@@ -209,6 +231,29 @@ class JavaApiBuilder : CodeGenerationAdapter.AbstractApiBuilder<TypeSpec.Builder
         }
     }
 
+    companion object {
+        private val elementGroups = linkedMapOf(
+            "Tasks" to setOf(
+                BpmnElementType.SERVICE_TASK, BpmnElementType.USER_TASK, BpmnElementType.RECEIVE_TASK,
```
Comment (emaarco) [inline bpmn-to-code-core/src/main/kotlin/io/github/emaarco/bpmn/adapter/outbound/codegen/builder/JavaApiBuilder.kt]: put each entry in a own line - not multiple in one. should be a default for code-style if not everything fits in one line. maybe we should add a .claude/rule for that

File: bpmn-to-code-core/src/main/kotlin/io/github/emaarco/bpmn/adapter/outbound/codegen/builder/JavaApiBuilder.kt [hunk 2]
```diff
@@ -139,6 +151,16 @@ class JavaApiBuilder : CodeGenerationAdapter.AbstractApiBuilder<TypeSpec.Builder
         override fun write(builder: TypeSpec.Builder, modelApi: BpmnModelApi) {
             val variablesBuilder = TypeSpec.classBuilder("Variables").addModifiers(PUBLIC, STATIC, FINAL)
             modelApi.model.variables.forEach { variable -> variablesBuilder.addField(createAttribute(variable)) }
+            modelApi.model.flowNodes
+                .filter { it.variables.isNotEmpty() }
+                .sortedBy { it.getRawName() }
+                .forEach { node ->
+                    val className = node.getRawName().toCamelCase()
+                    val nodeVarsBuilder = TypeSpec.classBuilder(className).addModifiers(PUBLIC, STATIC, FINAL)
+                    node.variables.sortedBy { it.getRawName() }
```
Comment (emaarco) [inline bpmn-to-code-core/src/main/kotlin/io/github/emaarco/bpmn/adapter/outbound/codegen/builder/JavaApiBuilder.kt]:
```suggestion
                    val sortedVariables = node.variables.sortedBy { it.getRawName() }
                    sortedVariables.forEach { nodeVarsBuilder.addField(createAttribute(it)) }
```

File: bpmn-to-code-core/src/main/kotlin/io/github/emaarco/bpmn/domain/utils/StringUtils.kt [hunk 3]
```diff
@@ -33,4 +34,13 @@ object StringUtils {
     fun String.removeExpressionSyntax(): String {
         return this.replace(Regex("[#\${}]"), "")
     }
+
+    /**
+     * Converts a string with underscores to CamelCase, capitalising the first letter of each segment.
+     * @sample toCamelCase will convert Activity_SendMail to ActivitySendMail
+     * @sample toCamelCase will convert StartEvent_RequestReceived to StartEventRequestReceived
+     */
+    fun String.toCamelCase(): String =
```
Comment (emaarco) [inline bpmn-to-code-core/src/main/kotlin/io/github/emaarco/bpmn/domain/utils/StringUtils.kt]: also something we could make a rule for / combining it with the one above. 
a rule for writing `.kt` files. 

Based on docs of claude-code.
That should always use function-based style instead of lambda if its multiline - to have a uniform style.

File: bpmn-to-code-core/src/test/resources/api/NewsletterSubscriptionProcessApiJava.txt [hunk 4]
```diff
@@ -36,6 +36,40 @@ public final class NewsletterSubscriptionProcessApi {
...
+
+      public static final String START_EVENT_SUBMIT_REGISTRATION_FORM = "StartEvent_SubmitRegistrationForm";
+
+      public static final String TIMER_AFTER_3_DAYS = "Timer_After3Days";
+
+      public static final String TIMER_EVERY_DAY = "Timer_EveryDay";
+    }
+
+    public static final class Containers {
+      public static final String CALL_ACTIVITY_ABORT_REGISTRATION = "CallActivity_AbortRegistration";
```
Comment (emaarco) [inline bpmn-to-code-core/src/test/resources/api/NewsletterSubscriptionProcessApiJava.txt]: would split containers & call-actvities. since container may be something like a eventBasedSubprocess as well. and call-activity is just a task referecning to some process (stil - make call-activity an own one)
